src/lm.jl

```diff
@@ -1,20 +1,35 @@
+# TODO: expand weights to include fweights, pweights, aweights
+# Example LmResp{Array{Float64,1}}(randn(5), randn(5), randn(5), randn(5))
 type LmResp{V<:FPVector} <: ModResp  # response in a linear model
     mu::V                                  # mean response
     offset::V                              # offset added to linear predictor (may have length 0)
     wts::V                                 # prior weights (may have length 0)
+    fweights::FrequencyWeights
+    pweights::ProbabilityWeights
+    aweights::AnalyticWeights
```

Comment:
Yes, we should store the weights internally as a single vector (maybe keep the existing `wts` field), and only keep track of the necessary additional information like the actual number of observations (which is either the sum of weights or the vector length depending on the weights type).